src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,450 @@
+                                     const instruction_ref loss_tensor, 
+                                     const instruction_ref weights, 
+                                     const std::string &reduction, 
+                                           bool has_ignore_index, 
+                                           bool has_weights) const    
+    {
+        instruction_ref final_loss_tensor = loss_tensor;
+
+        // Used for reductions
+        std::vector<size_t> loss_dims(loss_tensor->get_shape().ndim());
+        std::iota(loss_dims.begin(), loss_dims.end(), 0);
+
+        std::vector<size_t> weight_dims(weights->get_shape().ndim());
+        std::iota(weight_dims.begin(), weight_dims.end(), 0);
+
+        if(has_ignore_index or has_weights)
+        {
+            final_loss_tensor =
+                info.add_instruction(migraphx::make_op("mul"), loss_tensor, weights);
+        }
+
+        // Add reduction step after we're generated crossentropyloss tensor and rearragned weight
+        // scaling tensor
+        if(reduction == "mean" and has_weights)
+        {
+            final_loss_tensor =
+                info.add_instruction(migraphx::make_op("reduce_sum", {{"axes", loss_dims}}), final_loss_tensor);
+            auto reduced_weights = info.add_instruction(
+                migraphx::make_op("reduce_sum", {{"axes", weight_dims}}), weights);
+            final_loss_tensor =
+                info.add_instruction(migraphx::make_op("div"), final_loss_tensor, reduced_weights);
+        }
+        else if(reduction == "mean" and not has_weights)
+        {
+            final_loss_tensor = info.add_instruction(migraphx::make_op("reduce_mean", {{"axes", loss_dims}}),
+                                               final_loss_tensor);
+        }
+        else if(reduction == "sum")
+        {
+            final_loss_tensor =
+                info.add_instruction(migraphx::make_op("reduce_sum", {{"axes", loss_dims}}), final_loss_tensor);
+        }
+
+        return final_loss_tensor;
+    }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd */,
+                                       const onnx_parser& parser,
+                                       const onnx_parser::node_info& info,
+                                       const std::vector<instruction_ref>& args) const
+    {
+        // Get and handle attributes
+        auto reduction = get_reduction_param(info);
+
+        // Get and validate Inputs
+        auto scores = get_scores(args.at(0));
+        auto scores_shape = scores->get_shape();
+        auto labels  = get_labels(args.at(1), scores_shape);
+
+        // meta parameters based on input scores shape
+        size_t ndims = scores_shape.ndim();
+        size_t batch_size = scores_shape.lens().at(0);
+        size_t class_size = scores_shape.lens().at(1);
+        bool is_k_dim  = (ndims > 3);
+        bool sizes_mismatch = batch_size != class_size;
```

Comment:
why does this need to be handled seperately?

Reply:
This was so we didn't use the scatter_none case for mismatched batches and thus flowed into the gathernd logic